typed/remove_test.go

```diff
@@ -0,0 +1,646 @@
+		_NS(
+			_P("listOfMaps", _KBF("name", "q")),
+		),
+		`{"listOfMaps": [{"name": "a", "value": {"b":"x", "c":"y"}}, {"name": "d", "value": {"e":"z"}}]}`, // doesn't remove anything
+		`{"listOfMaps":null}`, // extract only the root type
+	}, {
+		// path with existant top-level but non-existant leaf element.
+		`{"listOfMaps": [{"name": "a", "value": {"b":"x", "c":"y"}}, {"name": "d", "value": {"e":"z"}}]}`,
+		_NS(
+			_P("listOfMaps", _KBF("name", "a"), "value", "q"),
+		),
+		`{"listOfMaps": [{"name": "a", "value": {"b":"x", "c":"y"}}, {"name": "d", "value": {"e":"z"}}]}`, // doesn't remove anything
+		`unparseable`, //`{"listOfMaps": [{"value": null}]}`, // unparseable because name cannot be missing
+	}, {
+		// paths with existant top-level but non-existant leaf element
+		`{"listOfMaps": [{"name": "a", "value": {"b":"x", "c":"y"}}, {"name": "d", "value": {"e":"z"}}]}`,
+		_NS(
+			_P("listOfMaps", _KBF("name", "a"), "name"),
+			_P("listOfMaps", _KBF("name", "a"), "value", "q"),
+		),
+		`unparseable`, // unparseable because remove cannot operate on a top-level element and a leaf within
+		`{"listOfMaps": [{"name":"a", "value": null}]}`, // parseable because value (map type) CAN be null
+	}, {
+		// path to root type
+		//`{"mapOfLists": {"b":["a","c"]}}`,
+		`{"mapOfLists": {"b":["a","c"], "d":["e", "f"]}}`,
+		_NS(
+			_P("mapOfLists"),
+		),
+		``,
+		`{"mapOfLists": {"b":["a","c"], "d":["e", "f"]}}`,
+	}, {
+		// path to a top-level element
+		`{"mapOfLists": {"b":["a","c"], "d":["e", "f"]}}`,
+		_NS(
+			_P("mapOfLists", "b"),
+		),
+		`{"mapOfLists": {"d":["e", "f"]}}`,
+		`{"mapOfLists": {"b":["a","c"]}}`,
+	}, {
+		// just one path to leaf element
+		`{"mapOfLists": {"b":["a","c"], "d":["e", "f"]}}`,
+		_NS(
+			_P("mapOfLists", "b", _V("a")),
+		),
+		`{"mapOfLists":{"b":["c"],"d":["e", "f"]}}`,
+		`{"mapOfLists":{"b":["a"]}}`,
+	}, {
+		// path to non-existant top-level element
+		`{"mapOfLists": {"b":["a","c"], "d":["e", "f"]}}`,
+		_NS(
+			_P("mapOfLists", "q"),
+		),
+		`{"mapOfLists": {"b":["a","c"], "d":["e", "f"]}}`,
+		`{"mapOfLists":null}`,
+	}, {
+		// path with existant top-level but non-existant leaf element
+		`{"mapOfLists": {"b":["a","c"], "d":["e", "f"]}}`,
+		_NS(
+			_P("mapOfLists", "b", _V("q")),
+		),
+		`{"mapOfLists": {"b":["a","c"], "d":["e", "f"]}}`,
+		`{"mapOfLists":{"b":null}}`,
+	}, {
+		// path with existant top-level but non-existant leaf element
+		`{"mapOfLists": {"b":null, "d":["e", "f"]}}`,
+		_NS(
+			_P("mapOfLists", "b"),
+		),
+		`{"mapOfLists": {"d":["e", "f"]}}`,
+		`{"mapOfLists":{"b":null}}`, // same output as previous case, but can be differentiated by input fieldpath.Set
+	}, {
+		// invalid path to just a leaf
+		`{"mapOfLists": {"b":["a","c"], "d":["e", "f"]}}`,
+		_NS(
+			_P(_V("a")),
+		),
+		`{"mapOfLists": {"b":["a","c"], "d":["e", "f"]}}`,
+		``,
+	}, {
+		// path to root type
+		`{"mapOfMaps": {"b":{"a":"x","c":"z"}, "d":{"e":"y", "f":"w"}}}`,
+		_NS(
+			_P("mapOfMaps"),
```

Comment:
we need to think about that too, that's not completely obvious to me

Reply:
This is exactly how Remove works currently. If you pass remove a fieldset with only the root element, it will remove everything and return an empty value, so I would Extract to work the same way in reverse, where passing a the root element will "extract" (i.e. keep) everything and return a copy of the original value.